frontend-react/src/components/FileHandlers/FileHandler.tsx

```diff
@@ -223,50 +262,79 @@ function FileHandler() {
                 },
             });
         }
+        const isFileSuccess =
+            eventData?.overallStatus === OverallStatus.VALID &&
+            !(eventData?.errorCount || eventData?.warningCount);
+        if (isFileSuccess) {
+            handleNextFileHandlerStep(3);
+        } else {
+            handleNextFileHandlerStep();
+        }
     };
 
-    const resetState = () => {
-        setFileContent("");
-        dispatch({ type: FileHandlerActionType.RESET });
-    };
-
-    const submitted = !!(reportId || errors.length || overallStatus);
-
-    // default to FILE messaging here, partly to simplify typecheck
-    const errorMessaging = errorMessagingMap[errorType || ErrorType.FILE];
-
-    let formLabel = "";
-    let successDescription = "";
-    if (selectedSchemaOption) {
-        formLabel = UPLOAD_PROMPT_DESCRIPTIONS[selectedSchemaOption.format];
-        successDescription = SUCCESS_DESCRIPTIONS[selectedSchemaOption.format];
-    }
-
     // Array containing only qualityFilterMessages that have filteredReportItems.
     const qualityFilterMessages = reportItems?.filter(
         (d) => d.filteredReportItems.length > 0
     );
 
+    // default to FILE messaging here, partly to simplify typecheck
+    const errorMessaging = errorMessagingMap[errorType || ErrorType.FILE];
+
     const hasQualityFilterMessages =
         destinations.length > 0 &&
         qualityFilterMessages &&
         qualityFilterMessages.length > 0;
 
-    const isFileSuccess =
-        (reportId || overallStatus === OverallStatus.VALID) &&
-        !hasQualityFilterMessages;
-
     if (isSenderSchemaOptionsLoading) {
-        return <FileHandlerSpinner message="Loading..." />;
+        return <FileHandlerSpinner message={<p>Loading...</p>} />;
     }
 
+    const handleOnSchemaChange = (schemaOption: SchemaOption) => {
+        dispatch({
+            type: FileHandlerActionType.SCHEMA_SELECTED,
+            payload: schemaOption,
+        });
+    };
+
+    const siteTooltip = () => (
```

Comment:
It's an anti-pattern to define a component inside another component: it creates a new definition _every_ render, and it bypasses any potential memoization and prop type checking.  Will update.